service/utils/rabbitmq.py

```diff
@@ -68,3 +68,40 @@ def hook(progress, loss):
         )
 
     return hook
+
+
+class OptimizeHook:
+    def __init__(self, job_id, total_possible_iterations):
+        connection = pika.BlockingConnection(
+            conn_config,
+        )
+        self.channel = connection.channel()
+        self.job_id = job_id
+        self.result = []
+        self.step = 0
+        self.total_possible_iterations = total_possible_iterations
+
+    def __call__(self, current_results):
+        self.step += 1
+        print(
+            "job_id: ",
+            self.job_id,
+            " progress: ",
+            self.step,
+            " current_results: ",
+            current_results.tolist(),
+            " total_possible_iterations: ",
+            self.total_possible_iterations,
+        )
```

Comment:
Not too sure on the intention of the print here, but
- Maybe use f-string to format instead of these comma separated args
- Maybe use the logging framework that should be set up instead of print

Reply:
I can just remove this tbh 
It was helpful for testing but I dont think its really helpful for moving forward. Good catch